sycl/doc/extensions/SPIRV/SPV_INTEL_variable_length_array.asciidoc

```diff
@@ -0,0 +1,246 @@
+:OpSaveMemoryINTEL_token          : pass:normal[5819]
+:OpRestoreMemoryINTEL_token       : pass:normal[5820]
+
+== Name Strings
+
+SPV_INTEL_variable_length_array
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+== Contributors
+
+- Alexey Sotkin, Intel +
+- Ben Ashbaugh, Intel +
+- Vyacheslav Zakharin, Intel +
+
+== Notice
+
+Copyright (c) 2020 Intel Corporation.  All rights reserved.
+
+== Status
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a
+feature for review and community feedback. When the feature matures, this
+specification may be released as a formal extension.
+
+
+Because the interfaces defined by this specification are not final and are
+subject to change they are not intended to be used by shipping software
+products. If you are interested in using this feature in your software product,
+please let us know!
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision           | A
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification,
+Version 1.2 Revision 1.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension allows to allocate local arrays whose number of elements is
+unknown at compile time. This is useful for implementing high-level language
+features like C99 variable length arrays.
+
+== Extension Name
+
+To use this extension within a SPIR-V module, the appropriate *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_variable_length_array"
+----
+
+== New Capabilities
+
+This extension introduces new capabilities:
+
+----
+VariableLengthArrayINTEL
+----
+
+== New Instructions
+
+Instructions added under the *VariableLengthArrayINTEL* capability:
+
+----
+OpVariableLengthArrayINTEL
+OpSaveMemoryINTEL
+OpRestoreMemoryINTEL
+----
+
+
+== Token Number Assignments
+
+[width="40%"]
+[cols="70%,30%"]
+|====
+|VariableLengthArrayINTEL   | {capability_token}
+|OpVariableLengthArrayINTEL | {OpVariableLengthArrayINTEL_token}
+|OpSaveMemoryINTEL          | {OpSaveMemoryINTEL_token}
+|OpRestoreMemoryINTEL       | {OpRestoreMemoryINTEL_token} 
+|====
+
+== Modifications to the SPIR-V Specification, Version 1.2
+
+=== Capabilities
+
+Modify Section 3.31, Capability, adding rows to the Capability table:
+
+[cols="1,10,4,4",options="header"]
+|====
+2+^| Capability ^| Implicitly Declares ^| Enabled by Extension
+
+| {capability_token}
+| *VariableLengthArrayINTEL*
+|
+| *SPV_INTEL_variable_length_array*
+
+|====
+
+=== Instructions
+
+Modify Section 3.36.8. Memory Instructions, adding to the end of the list of
+instructions:
+
+[cols="1,1,3*3",width="100%"]
+|=====
+4+|[[OpVariableLengthArrayINTEL]]*OpVariableLengthArrayINTEL* +
+ +
+Allocate a runtime-sized array in the function memory, resulting in a pointer to
```

Comment:
Is "in the function memory" the definition that we want here?  Would it be better to say "function storage class"?

Reply:
Right, I'm not sure about "function memory". What about:
"Allocate a runtime-sized array in memory, resulting in a pointer to it, which can be used with *OpLoad* and *OpStore*.
The memory is allocated in *Function* storage class and is uninitialized." ?

Reply:
Yes, I think that's an improvement and is fine with me.